applications/nrf5340_audio/src/bluetooth/le_audio_bis_headset.c

```diff
@@ -119,9 +129,14 @@ static bool bitrate_check(const struct bt_codec *codec)
 
 static bool adv_data_parse(struct bt_data *data, void *user_data)
 {
+	struct bt_name *bis_name = (struct bt_name *)user_data;
+
 	if (data->type == BT_DATA_BROADCAST_NAME && data->data_len) {
-		memcpy((char *)user_data, data->data, data->data_len);
-		return false;
+		if (data->data_len <= bis_name->size) {
+			memcpy(bis_name->name, data->data, data->data_len);
+			bis_name->name[data->data_len] = '\0';
```

Comment:
Seems like this will overwrite the last character?

Reply:
I believe the array index is 0-to-(data->data_len - 1) and hence this actually writes to the last character position plus one.

Reply:
I think this is correct

Reply:
I have to prioritize some other issues, so I have not looked, but as long as we don't overflow this is ok.

Reply:
That is a good point I think the if should be < rather than <=

Reply:
Good point the if should be < rather than <= otherwise the name string will overflow.
The size give is the size of the name plus the terminator character as it is an array of strings. If it wasn't then we would have to add the terminator ourselves.

Reply:
Good question, but the size of the bis_name is just the number of characters but is stored in a chat array of this size + 1 as it is a string. So it is all good here!

Reply:
That is a good point. It should not be needed since name[]/bis_name is initialized to all "\0"s

Reply:
True, but I would say this is good practice to make the function robust or there should be a comment here to the effect that the string array passed into this function is assumed to be filled with terminator characters. It maybe better to make no assumptions about the string and in the callback do not initialize it.

Reply:
Hmmm, I also agree with you @gWacey. It is not optimal that the function assumes the array being initialized to that.
It is kind of in the gray area, since it is a local function only used in this file

Reply:
I added it back